source/common/stats/stats_impl.h

```diff
@@ -61,11 +74,56 @@ class TagProducerImpl : public TagProducer {
   std::string produceTags(const std::string& metric_name, std::vector<Tag>& tags) const override;
 
 private:
+  friend class DefaultTagRegexTester;
+
+  /**
+   * Adds a TagExtractor to the collection of tags, tracking prefixes to help make
+   * produceTags run efficiently by trying only extractors that have a chance to match.
+   * @param extractor TagExtractorPtr the extractor to add.
+   */
+  void addExtractor(TagExtractorPtr extractor);
+
+  /**
+   * Adds all default extractors matching the specified tag name. In this model,
+   * more than one TagExtractor can be used to generate a given tag. The default
+   * extractors are specified in common/config/well_known_names.cc.
+   * @param name absl::string_view the extractor to add.
+   */
+  void addExtractorsMatching(absl::string_view name);
+
+  /**
+   * Roughly estimate the size of the vectors.
+   * @param config const envoy::config::metrics::v2::StatsConfig& the config.
+   */
   void reserveResources(const envoy::config::metrics::v2::StatsConfig& config);
-  void addDefaultExtractors(const envoy::config::metrics::v2::StatsConfig& config,
-                            std::unordered_set<std::string>& names);
 
-  std::vector<TagExtractorPtr> tag_extractors_;
+  /**
+   * Adds the names of all default extractors into a string-set for dup-detection
+   * against new stat names specified in the configuration.
+   * @param config const envoy::config::metrics::v2::StatsConfig& the config.
+   * @return names std::unordered_set<std::string> the set of names to populate
+   */
+  std::unordered_set<std::string>
+  addDefaultExtractors(const envoy::config::metrics::v2::StatsConfig& config);
```

Comment:
Does this actually "add" default extractors somewhere or just put them into the returned set that was otherwise empty? The name is a bit misleading. It also seems weird to say the return is "the set of names to populate" - it's already populated once returned, right?

Reply:
yes, the function description was inaccurate.  Sorry about that, & good catch.  Updated the description to:
```
  /**
   * Adds all default extractors from well_known_names.cc into the
   * collection.  Returns a set of names of all default extractors
   * into a string-set for dup-detection against new stat names
   * specified in the configuration.
   * @param config const envoy::config::metrics::v2::StatsConfig& the config.
   * @return names std::unordered_set<std::string> the set of names to populate
   */
```